### PR TITLE
fix: create user with password broken [DET-6852]

### PR DESCRIPTION
**Pull request by varlogtim**
## Test Plan

Create a user with a password via the REST API.

```
# Set a curl'able master URL:
export DET_MASTER=http://localhost:8080

# Login as admin
export TOKEN=$(curl -X POST -d '{"username": "admin", "password": ""}' ${DET_MASTER}/api/v1/auth/login | jq '.token' | sed 's/"//g')

# Create a user with password:
curl -H "Authorization: Bearer ${TOKEN}" -H  "accept: application/json" -X POST "${DET_MASTER}/api/v1/users" -d '{"user": {"username": "mytestuser", "admin": false, "active": true}, "password": "asdf"}'

# Try to login as the user using the password "asdf"
det user login mytestuser
```

### CODE REVIEW

**Review by stoksc — APPROVED** [review]
approved by @mapmeld should look too

File: master/pkg/model/test_utils.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func ExperimentModel(opts ...ExperimentModelOption) *Experiment {
 				RawMaxLength: &maxLength,
 			},
 		},
-		RawEntrypoint: &expconf.Entrypoint{RawEntrypoint: "model_def:SomeTrialClass"},
+		RawEntrypoint:      &expconf.Entrypoint{RawEntrypoint: "model_def:SomeTrialClass"},
```
Comment (mapmeld) [inline master/pkg/model/test_utils.go:40]: I don't know if this is picked up by lint, but I think you can remove this change and keep the file unchanged

**Comment by mapmeld** [comment]
I think I understand, you're creating the user from the command line in one POST request.

[The CLI also calls `POST /api/v1/users`](https://github.com/determined-ai/determined/blob/master/harness/determined/cli/user.py#L172) but without a password so we hadn't checked this issue before

File: master/pkg/model/test_utils.go (hunk 1 above)
Comment (varlogtim) [reply master/pkg/model/test_utils.go:40]: I feel like this was caused by lint as I didn't make it.

**Comment by varlogtim** [comment]
> I think I understand, you're creating the user from the command line in one POST request.
> 
> [The CLI also calls `POST /api/v1/users`](https://github.com/determined-ai/determined/blob/master/harness/determined/cli/user.py#L172) but without a password so we hadn't checked this issue before

More details: an OSS user asked why their POST user with password didn't work, I glanced at the endpoint and it looked like it was doing stuff with the password and should work, I verified that it did not work, now we are here.

I am thinking the person who made the CLI used the same kind of principals as most linux tools, i.e., always prompt for passwords.

**Review by mapmeld — APPROVED** [review]

File: master/pkg/model/test_utils.go (hunk 1 above)
Comment (stoksc) [reply master/pkg/model/test_utils.go]: this is probably lint not getting run with tags properly or something. i'll fix.

**Pull request merged**